### ISSUE TITLE
Bind to specific IP

### ISSUE THREAD

**Issue by auro1** [phpseclib/phpseclib#742]
Hello there.

I'm on a virtual server with multiple IP's.
I need a way to bind the phpseclib to a specific IP 

I can't seem to figure out how to do this with phpseclib - this is how I would open a socket bound to a *specific* ip
```
$sock = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
socket_bind($sock, '192.168.1.100'); // Bind to IP 192.168.1.100
socket_connect($sock, 'example.net', 80);
```

**Comment by terrafrost** [comment]
#746 should let you now do this. eg.

```php
// http://us3.php.net/manual/en/context.socket.php
$opts = array(
    'socket' => array(
        'bindto' => '192.168.1.100:0',
    ),
);
$context = stream_context_create($opts);
$socket = stream_socket_client('tcp://example.net:80', $errno, $errstr, ini_get('default_socket_timeout'), STREAM_CLIENT_CONNECT, $context);

$ssh = new Net_SSH2($socket); 
```

**Comment by auro1** [comment]
Wow. That's awesomely fast! Looking forward to seeing it merged :+1:

**Comment by bantu** [comment]
Fixed by #746
